src/Http/Client/Adapter/Curl.php

```diff
@@ -110,9 +110,10 @@ public function buildOptions(RequestInterface $request, array $options): array
         }
 
         $body = $request->getBody();
-        $body->rewind();
-        $out[CURLOPT_POSTFIELDS] = $body->getContents();
-
+        if (!empty($body->getContents())) {
+            $body->rewind();
+            $out[CURLOPT_POSTFIELDS] = $body->getContents();
```

Comment:
Retrieve (after rewind) contents once, then check `!== ''`.

Your `if` now checks non-rewinded contents and might be at end of stream. Also `empty` is useful on variables, not function calls.

Reply:
I could make these changes, I see the other proposed change by @markstory , I take it these are two different solutions.